contrib/babelfishpg_tsql/sql/fts_contains_rewrite.sql

```diff
@@ -54,17 +55,19 @@ BEGIN
 
   -- Split the phrase into an array of words
   FOREACH word IN ARRAY regexp_split_to_array(phrase COLLATE "C", '\s+' COLLATE "C") COLLATE "C" LOOP
-    -- prefix term not supported
-    IF (word COLLATE C) SIMILAR TO ('%\*' COLLATE C)  THEN
-      RAISE EXCEPTION 'Prefix term not supported';
-    END IF;
     -- Append the word to the joined_text variable
     joined_text := joined_text || word || '<->';
   END LOOP;
 
   -- Remove the trailing "<->" from the joined_text
   joined_text := substring(joined_text COLLATE "C", 1, length(joined_text) - 3) COLLATE "C";
 
+  -- If the last word has the form 'lastword*', then this is a prefix term (the last word cannot be '*')
+  -- We need to rewrite the last word into 'lastword:*'
+  IF ((joined_text COLLATE "C") SIMILAR TO ('%\*' COLLATE C)) AND (NOT (joined_text COLLATE "C") SIMILAR TO ('%<->\*' COLLATE C)) AND (NOT (joined_text COLLATE "C") SIMILAR TO ('\*' COLLATE C)) THEN
```

Comment:
Are we sure all the similar to pattern match has a test cases ?

Reply:
Not really. This is an expedient solution. The ideal way is to parse those search terms by writing parser rules in a parser generator. However, current ANTLR parser/Bison parser do not fit our needs to parse it, because the search term is lexed as a single string constant token by the lexer and there is no way for a parser to look into a single token. To parse the search term we will need to add another parser generator specifically for full text search which requires substantial change to the existing software architecture.

Reply:
I just added test cases for the patter match